samples/dotnet/kernel-syntax-examples/Example12_Planning.cs

```diff
@@ -64,6 +65,40 @@ private static async Task ConditionalSampleAsync()
         await ExecutePlanAsync(kernel, planner, originalPlan, 20);
     }
 
+    private static async Task WhileConditionalSampleAsync()
+    {
+        Console.WriteLine("======== Planning - While Loop Conditional flow example ========");
+        var kernel = InitializeKernelAndPlanner(out var planner);
+
+        // Load additional skills to enable planner to do non-trivial asks.
+        string folder = RepoFiles.SampleSkillsPath();
+        kernel.ImportSemanticSkillFromDirectory(folder, "FunSkill");
+        kernel.ImportSkill(new MathSkill());
+        kernel.ImportSkill(new TimeSkill());
+
+        var originalPlan = await kernel.RunAsync(
+            @"Start with a X number equals to the current minutes of the clock and remove 20 from this number until it becomes 0. After that tell me a math style joke where the input is X number + ""bananas""",
+            planner["CreatePlan"]);
+
+        /*
+        <goal>
+        Start with a X number equals to the current minutes of the clock and remove 20 from this number until it becomes 0. After that tell me a math style joke where the input is X number + "bananas"
+        </goal>
+        <plan>
+          <function._GLOBAL_FUNCTIONS_.Minute setContextVariable="X_NUMBER"/>
+          <while condition="$X_NUMBER greaterthan 0">
+            <function._GLOBAL_FUNCTIONS_.Subtract input="$X_NUMBER" Amount="20" setContextVariable="X_NUMBER"/>
+          </while>
+          <function.FunSkill.Joke input="$X_NUMBER bananas" style="Math" appendToResult="RESULT__JOKE"/>
```

Comment:
Does this work now? Did you change this in this PR or previous?

Reply:
Yes, this one works in this PR, `while` condition. The other has another example `IfConditionaSampleAsync` above.